test/test_autograd.py

```diff
@@ -9019,19 +9070,38 @@ def train_fn_grad(x):
         # be accumulate to the same place and should be the same
         self._run_py_multithread_fn(train_fn_grad, (x,))
 
-    def test_multithread_saved_tensors_hooks(self):
-        def pack(x):
-            warnings.warn("pack")
-            return x
+    def test_multi_grad_hooks(self):
+        # Multihooks should behave independently per execution of backward
```

Comment:
I think a good way to test this is to use a custom Function that raise an error.
That way you will leave a multi-hook partially applied.